azkaban-webserver/azkaban-cognito-usermanager/src/main/java/uk/gov/dwp/dataworks/azkaban/authentication/CognitoUserManager.java

```diff
@@ -294,7 +296,10 @@ private User getCognitoUser(final String token) throws UserManagerException {
     }
     Map<String, Claim> claims = decodedToken.getClaims();
     User user = null;
-    user = new User(claims.get("cognito:username").asString());
+    String username = claims.get("cognito:username").asString();
```

Comment:
Username needs special consideration. its either (preferred_username | username) suffixed with the first 3 chars of the sub.

Reply:
Should be sorted now